diff --git a/dist/private/utils/runAfterTransition.js b/dist/private/utils/runAfterTransition.js
--- a/dist/private/utils/runAfterTransition.js
+++ b/dist/private/utils/runAfterTransition.js
@@ -37,7 +37,7 @@
             // event. This enables us to handle when the node is deleted from the document while it is transitioning.
             // In that case, the cancel event would have nowhere to bubble to so we need to handle it directly.
             eventTarget.addEventListener('transitioncancel', onTransitionEnd, {
-                once: true
+                once: false
             });
         }
         transitions.add(e.propertyName);
diff --git a/dist/private/utils/runAfterTransition.mjs b/dist/private/utils/runAfterTransition.mjs
--- a/dist/private/utils/runAfterTransition.mjs
+++ b/dist/private/utils/runAfterTransition.mjs
@@ -37,7 +37,7 @@
             // event. This enables us to handle when the node is deleted from the document while it is transitioning.
             // In that case, the cancel event would have nowhere to bubble to so we need to handle it directly.
             eventTarget.addEventListener('transitioncancel', onTransitionEnd, {
-                once: true
+                once: false
             });
         }
         transitions.add(e.propertyName);
diff --git a/dist/private/utils/runAfterTransition.cjs b/dist/private/utils/runAfterTransition.cjs
--- a/dist/private/utils/runAfterTransition.cjs
+++ b/dist/private/utils/runAfterTransition.cjs
@@ -43,7 +43,7 @@
             // event. This enables us to handle when the node is deleted from the document while it is transitioning.
             // In that case, the cancel event would have nowhere to bubble to so we need to handle it directly.
             eventTarget.addEventListener('transitioncancel', onTransitionEnd, {
-                once: true
+                once: false
             });
         }
         transitions.add(e.propertyName);
